custom/staking/src/lib.rs

```diff
@@ -933,8 +965,8 @@ decl_module! {
                 Err(Error::<T>::AlreadyPaired)?
             }
 
-            // reject a bond which is considered to be _dust_.
-            if value < T::Currency::minimum_balance() {
+            // Reject a bond if value is _dust_ or less than min bond threshold
+            if value < T::Currency::minimum_balance() || value < <MinimumBondThreshold<T>>::get() {
```

Comment:
If we add that check here then minimum_balance check will also be applied for nominators as well. I know at the time of calling the `bond()` function we don't know whether the controller will be a `nominator` or it will be a `potential validator`. It is more of a business question if we want to impose the restriction on nominators as well or not. 

Other way around to solve the problem is not to put that check here. And Add that check at `validate()` function which will dynamically check the bond balance and if it more than the minimum balance then it will good otherwise validate transaction will fail.

Reply:
The minimum should just be on validators, not nominators (who aren't validators). So maybe this should be checked each session or era when we do the other validator checks?

Reply:
Currently we are checking at the beginning of each era. Validators who don't have a min balance are left out in the next era.

Reply:
As @satyamakgec says, with the logic here, won't we be restricting non-validator nominators to the same minimum bond?

Reply:
Yeah @adamdossa , we are restricting the nominators here as well. The easy fix to remove that check from here and add that check in `validate()` function where a controller shows its desire to be a potential validator and at that function, we can compare the bond balance with minimum balance if it is less then we will not add that controller into the potential validator list and emit an error like `Insufficient bonded balance`